ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/services/FragmentComparator.java

```diff
@@ -0,0 +1,56 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.TreeRelation;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.Fragment;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class FragmentComparator {
+	public static Stream<Fragment> sortFragments(Stream<Fragment> fragmentStream) {
+		List<Fragment> fragments = fragmentStream.toList();
+
+		var firstElement = fragments.stream()
+				.filter(fragment -> hasNoConnections(fragment, fragments))
+				.findFirst()
+				.orElseThrow();
+
+		var map = fragments.stream()
+				.filter(fragment -> !fragment.getRelations().isEmpty())
+				.collect(Collectors.toMap(Fragment::getFragmentIdString, fragment -> fragment.getRelations()
+						.stream()
+						.findFirst()
```

Comment:
It may be useful to add our own stream collector in the server that allows us to collect the only single result and throws/or logs an error if there is no result or more than one result. Similar like we have in hibernate.

Reply:
point for dev session